### PR TITLE
fix: script imports for tpescript 

### PR DESCRIPTION
**Pull request by Sembauke**

### CODE REVIEW

**Review by Nirajn2311 — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Could you elaborate a bit on the purpose of these changes?  For the most part it looks like they should be changes to strip-comments, rather than this repo.

**Comment by Sembauke** [comment]
> 



> Could you elaborate a bit on the purpose of these changes? For the most part it looks like they should be changes to strip-comments, rather than this repo.

It wouldn't make much sense to keep strip-comments in it's own repo, it is only used with the curriculum helpers.  We also wanted to convert it from a commonJS module to a ECMA module so that it could be easily implemented with script imports without doing a lot of fancy typescript configurations.

**Comment by Sembauke** [comment]
Hi @ojeytonwilliams 
I agree we should not leave out the extra users of the strip-comments library. But I think this is the way to go for the short term. Let's open an issue for it on the strip comments library when this is merged in.

**Comment by Sembauke** [comment]
Babel is still needed to run Jest with ESM modules.

**Comment by ojeytonwilliams** [comment]
> Babel is still needed to run Jest with ESM modules.

Ah, right, I completely missed that.  Thanks!

File: .babelrc [hunk 1]
```diff
@@ -7,5 +7,6 @@
       }
     }]
     
-  ]
+  ],
+  //"plugins": ["babel-plugin-add-import-extension"]
```
Comment (ojeytonwilliams) [inline .babelrc:11]:
```suggestion
  ]
```

File: lib/strip.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ * @param  {Object} `options` optional options, passed to [extract-comments][extract-comments]
+ * @option {Boolean} [options] `line` if `false` strip only block comments, default `true`
+ * @option {Boolean} [options] `block` if `false` strip only line comments, default `true`
+ * @option {Boolean} [options] `keepProtected` Keep ignored comments (e.g. `/*!` and `//!`)
+ * @option {Boolean} [options] `preserveNewlines` Preserve newlines after comments are stripped
+ * @return {String} modified input
+ * @api public
+ */
+
+export const strip = ((input, options) => {
```
Comment (ojeytonwilliams) [inline lib/strip.ts:34]:
```suggestion
export const strip = ((input, options?) => {
```

File: lib/strip.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ * // => 'const foo = "bar";// this is a comment'
+ * ```
+ * @name  .block
+ * @param  {String} `input` string from which to strip comments
+ * @param  {Object} `options` pass `opts.keepProtected: true` to keep ignored comments (e.g. `/*!`)
+ * @return {String} modified string
+ * @api public
+ */
+
+strip.block= (input, options)=> {
```
Comment (ojeytonwilliams) [inline lib/strip.ts:55]:
```suggestion
strip.block= (input, options?)=> {
```

File: lib/strip.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+ * // => 'const foo = "bar";\n/* me too *\/'
+ * ```
+ * @name  .line
+ * @param  {String} `input` string from which to strip comments
+ * @param  {Object} `options` pass `opts.keepProtected: true` to keep ignored comments (e.g. `//!`)
+ * @return {String} modified string
+ * @api public
+ */
+
+strip.line = (input, options) => {
```
Comment (ojeytonwilliams) [inline lib/strip.ts:75]:
```suggestion
strip.line = (input, options?) => {
```

File: lib/strip.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+ * // => '//! first comment\nfoo; '
+ * ```
+ * @name .first
+ * @param {String} `input`
+ * @param {Object} `options` pass `opts.keepProtected: true` to keep comments with `!`
+ * @return {String}
+ * @api public
+ */
+
+strip.first = (input, options) => {
```
Comment (ojeytonwilliams) [inline lib/strip.ts:96]:
```suggestion
strip.first = (input, options?) => {
```

File: lib/parse.ts [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+const constants = {
+  ESCAPED_CHAR_REGEX: /^\\./,
+  QUOTED_STRING_REGEX: /^(['"`])((?:\\\1|[^\1])*?)(\1)/,
+  NEWLINE_REGEX: /^\r*\n/,
+};
+
+
+
+export const parse = (input: string, options: Partial<options> = {}) => {
```
Comment (ojeytonwilliams) [inline lib/parse.ts:15]:
```suggestion
export const parse = (input: string, options?: Partial<options> = {}) => {
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
This is looking good.  I just found a few minor things here and there:

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**